### PR TITLE
Adding documentation for transform_if to guide

### PR DESCRIPTION
**Pull request by danhoeflinger**
Adds documentation of transform_if (https://github.com/oneapi-src/oneDPL/pull/276) to additional algorithms.

### CODE REVIEW

File: documentation/library_guide/parallel_api/additional_algorithms.rst [hunk 1]
```diff
@@ -86,3 +86,26 @@ header.  All algorithms are implemented in the ``oneapi::dpl`` namespace.
     values: ['a', 'b', 'c', 'd', 'e', 'f']
     output_keys:   [0,    0,   3,   3,   4,   5]
     output_values: ['c', 'f', 'a', 'e', 'd', 'b']
+
+* ``transform_if``: performs a transform on the elements of the input sequence(s) and stores the result into
```
Comment (dcbenito) [inline documentation/library_guide/parallel_api/additional_algorithms.rst:90]: My suggestion:

* ``transform_if``: performs a transform on the input sequence(s) elements and stores the result into the corresponding position in the output sequence at each position for which the predicate applied to the element(s) evaluates to ``true``. If the predicate evaluates to ``false``, the transform is not applied for the elements(s), and the output sequence's corresponding position is left unmodified. There are two overloads of this function, one for a single input sequence with a unary transform and a unary predicate, and another for two input sequences and a binary transform and a binary predicate.
  Unary example:

    unary predicate: [](auto i){return i % 2 == 0;} // is even
    unary transform: [](auto i){return i * 2;}      // double element
    input sequence:           [0, 1, 2, 3, 3, 3, 4, 4, 7, 6]
    original output sequence: [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
    final output sequence:    [0, 8, 4, 6, 5, 4, 8, 8, 1, 12]

  Binary example:
  
    binary predicate: [](auto a, auto b){return a == b;} // are equal
    unary transform:  [](auto a, auto b){return a + b;}  // sum values
    input sequence1:           [0, 1, 2, 3, 3, 3, 4, 4, 7, 6]
    input sequence2:           [5, 1, 3, 4, 3, 3, 4, 4, 7, 9]
    original output sequence:  [9, 9, 9, 9, 9, 9, 9, 9, 9, 9]
    final output sequence:     [9, 2, 9, 9, 6, 6, 8, 8, 14, 9]

The formatting in this comment isn't working the way that I would like, but we should be able to format the examples like the previous entries.

**Review by dcbenito — CHANGES_REQUESTED** [review]

File: documentation/library_guide/parallel_api/additional_algorithms.rst (hunk 1 above)
Comment (danhoeflinger) [reply documentation/library_guide/parallel_api/additional_algorithms.rst]: I have applied what I understand to be the suggested changes:

- Wording changes in the description
- Adding a linebreak to the examples
- Adding " example" to each example heading

Please check again if you would, and confirm that these are the intended suggestions.  Its unclear to me if you wanted any other changes to the formatting (for instance in the whitespace of the example content).

File: documentation/library_guide/parallel_api/additional_algorithms.rst [hunk 2]
```diff
@@ -86,3 +86,28 @@ header.  All algorithms are implemented in the ``oneapi::dpl`` namespace.
...
+
+
+    Binary example:
+
+    binary predicate: [](auto a, auto b){return a == b;} // are equal
+    unary transform:  [](auto a, auto b){return a + b;}  // sum values
+    input sequence1:           [0, 1, 2, 3, 3, 3, 4, 4, 7, 6]
+    input sequence2:           [5, 1, 3, 4, 3, 3, 4, 4, 7, 9]
+    original output sequence:  [9, 9, 9, 9, 9, 9, 9, 9, 9, 9]
+    final output sequence:     [9, 2, 9, 9, 6, 6, 8, 8, 14, 9]
```
Comment (dcbenito) [inline documentation/library_guide/parallel_api/additional_algorithms.rst:113]:
```suggestion
* ``transform_if``: performs a transform on the input sequence(s) elements and stores the result into the
  corresponding position in the output sequence at each position for which the predicate applied to the 
  element(s) evaluates to ``true``. If the predicate evaluates to ``false``, the transform is not applied for
  the elements(s), and the output sequence's corresponding position is left unmodified. There are two overloads
  of this function, one for a single input sequence with a unary transform and a unary predicate, and another
  for two input sequences and a binary transform and a binary predicate.

  Unary example::

    unary predicate: [](auto i){return i % 2 == 0;} // is even
    unary transform: [](auto i){return i * 2;}      // double element
    input sequence:           [0, 1, 2, 3, 3, 3, 4, 4, 7, 6]
    original output sequence: [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
    final output sequence:    [0, 8, 4, 6, 5, 4, 8, 8, 1, 12]


  Binary example::

    binary predicate: [](auto a, auto b){return a == b;} // are equal
    unary transform:  [](auto a, auto b){return a + b;}  // sum values
    input sequence1:           [0, 1, 2, 3, 3, 3, 4, 4, 7, 6]
    input sequence2:           [5, 1, 3, 4, 3, 3, 4, 4, 7, 9]
    original output sequence:  [9, 9, 9, 9, 9, 9, 9, 9, 9, 9]
    final output sequence:     [9, 2, 9, 9, 6, 6, 8, 8, 14, 9]
```

**Review by dcbenito — CHANGES_REQUESTED** [review]

File: documentation/library_guide/parallel_api/additional_algorithms.rst (hunk 2 above)
Comment (dcbenito) [reply documentation/library_guide/parallel_api/additional_algorithms.rst]: Looks good!

**Review by dcbenito — APPROVED** [review]

**Review by timmiesmith — APPROVED** [review]
LGTM

**Pull request merged**